File: 백준/17413/solution.swift
let s = readLine()!

var tagSwitch: Bool = false 
var tag: String = ""
var word: String = ""
var substring: String = ""
var answer = [String]()
for c in s {
    if tagSwitch == true {
        if c == ">" {
            tag += ">"
            answer.append(tag)
            tag = ""
            tagSwitch = false
        } else {
            tag += String(c)
        }
    } else {
        if c == "<" {
            if !substring.isEmpty {
                word += String(substring.reversed())
                substring = ""
            }
            //reverse temp
            if !word.isEmpty {
                if word.last! == " "{
                    word.removeLast()
                }
                answer.append(word)
                word = ""
            }
            tagSwitch = true
            tag += "<"
            continue
        } else {
            if c == " " {
                if !substring.isEmpty {
                    word += String(substring.reversed())
                    word += " "
                    substring = ""
                }
            }
            substring += String(c)
        }
    }
}
if !substring.isEmpty {
    word += String(substring.reversed())
}
if !word.isEmpty {
    answer.append(word)
}
print(answer)

File: 백준/17413/solution.py
import sys
input = sys.stdin.readline
word = list(input().rstrip())
i = start = 0

while i < len(word):
    # <
    if word[i] == '<':
        while word[i] != '>':
            i += 1
    # word
    if word[i].isalnum():
        start = i
        while i < len(word) and word[i].isalnum():
            i += 1
        temp = word[start:i][::-1]
        word[start:i] = temp
    else:
        i += 1
print(''.join(word))